src/main/java/com/teragrep/jai_02/keystore/KeyStoreAccess.java

```diff
@@ -0,0 +1,106 @@
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jai_02.keystore;
+
+import javax.crypto.SecretKey;
+import java.security.*;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Map;
+import java.util.HashMap;
+
+public class KeyStoreAccess {
+    protected final String keyStorePath;
+    protected final char[] keyStorePassword;
+    private final KeyFactory keyFactory;
+    private final Map<String, String> userToAliasMapping;
+    protected final KeyStore keyStore;
+
+    public KeyStoreAccess(final String keyStorePath, final char[] keyStorePassword) {
+        this(keyStorePath, keyStorePassword, new KeyFactory());
+    }
+    public KeyStoreAccess(final String keyStorePath, final char[] keyStorePassword, final KeyFactory keyFactory) {
+        this.keyFactory = keyFactory;
+        this.keyStorePassword = keyStorePassword;
+        this.keyStorePath = keyStorePath;
+        this.userToAliasMapping = new HashMap<>();
+        this.keyStore = new KeyStoreFactory(keyStorePath, keyStorePassword).build();
+    }
+
+    public SecretKey loadKey(final String username) throws UnrecoverableEntryException, KeyStoreException, InvalidKeyException {
+        // TODO 3 create a cache of requests -> success/fail
+
+        // get alias mapping
+        final String alias;
+        if (userToAliasMapping.containsKey(username)) {
```

Comment:
will this work if the KeyStore-file is not created here but an already existing one?

Reply:
add UserToAliasMapping object to initialize map with existing keyStore